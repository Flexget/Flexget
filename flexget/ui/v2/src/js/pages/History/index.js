import { connect } from 'react-redux';
import { request } from 'utils/actions';
import { GET_HISTORY } from 'store/history/actions';
import History from './History';

function mapDispatchToProps(dispatch) {
  return {
    getHistory: data => dispatch(request(GET_HISTORY, data)),
  };
}

export default connect(null, mapDispatchToProps)(History);
export { History };
