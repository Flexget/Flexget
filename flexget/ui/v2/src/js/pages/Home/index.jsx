import React from 'react';
import InfoCard from './InfoCard';

const Home = () => <InfoCard />;

export default Home;
