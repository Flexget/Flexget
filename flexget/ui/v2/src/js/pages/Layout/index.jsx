import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Logo from 'pages/Layout/Logo';
import Navbar from 'pages/Layout/Navbar';
import SideNav from 'pages/Layout/SideNav';
import LoadingBar from 'common/LoadingBar';
import ErrorStatus from 'common/ErrorStatus';
import InfoStatus from 'common/InfoStatus';
import {
  Wrapper,
  Main,
  Header,
  LogoWrapper,
  Nav,
  SideBar,
  Content,
} from './styles';

class Layout extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  static defaultProps = {
    loggedIn: false,
  };

  state = {
    sideBarOpen: (window.matchMedia && !!window.matchMedia('(min-width: 600px)').matches) || false,
  };

  toggleSideBar = () => {
    this.setState({
      sideBarOpen: !this.state.sideBarOpen,
    });
  }

  render() {
    const { children } = this.props;
    const { sideBarOpen } = this.state;

    return (
      <Wrapper>
        <Header>
          <LogoWrapper>
            <Logo sideBarOpen={sideBarOpen} />
          </LogoWrapper>
          <Nav>
            <Navbar toggle={this.toggleSideBar} />
            <LoadingBar />
          </Nav>
        </Header>
        <Main>
          <SideBar>
            <SideNav
              sideBarOpen={sideBarOpen}
              toggle={this.toggleSideBar}
            />
          </SideBar>
          <Content open={sideBarOpen}>
            { children }
          </Content>
          <ErrorStatus />
          <InfoStatus />
        </Main>
      </Wrapper>
    );
  }
}

export default Layout;
