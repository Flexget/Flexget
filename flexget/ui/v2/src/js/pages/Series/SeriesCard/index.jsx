import React from 'react';
import PropTypes from 'prop-types';
import Card, { CardContent } from 'material-ui/Card';
import Typography from 'material-ui/Typography';
import blank from 'images/blank-banner.png';
import {
  Media,
  Image,
} from './styles';

const SeriesCard = ({ show }) => (
  <Card>
    <Media
      image={show.lookup.tvdb.banner ? `/api/cached?url=${show.lookup.tvdb.banner}` : ''}
      title={show.lookup.tvdb.series_name}
    >
      <Image src={blank} alt="" />
    </Media>
    <CardContent>
      <Typography type="headline" component="h2">
        {show.name}
      </Typography>
    </CardContent>
  </Card>
);

SeriesCard.propTypes = {
  show: PropTypes.shape({
    name: PropTypes.string,
    lookup: PropTypes.shape({
      tvdb: PropTypes.shape({
        banner: PropTypes.string,
        series_name: PropTypes.string,
      }),
    }),
  }).isRequired,
};

export default SeriesCard;
