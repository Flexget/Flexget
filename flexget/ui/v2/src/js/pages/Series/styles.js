import styled from 'react-emotion';
import Grid from 'material-ui/Grid';

export const GridCard = styled(Grid)`
  width: 100vw;
`;
