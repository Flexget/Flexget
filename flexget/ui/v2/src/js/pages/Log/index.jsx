import React from 'react';
import { PaperWrapper } from 'common/styles';
import Header from './Header';
import LogTable from './LogTable';
import { TableWrapper } from './styles';

const LogPage = () => (
  <PaperWrapper elevation={4}>
    <Header />
    <TableWrapper>
      <LogTable />
    </TableWrapper>
  </PaperWrapper>
);

export default LogPage;
