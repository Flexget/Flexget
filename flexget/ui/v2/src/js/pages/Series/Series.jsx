import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Grid from 'material-ui/Grid';
import SeriesCard from './SeriesCard';
import { GridCard } from './styles';

class SeriesPage extends Component {
  static propTypes = {
    shows: PropTypes.arrayOf(PropTypes.object).isRequired,
    getShows: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.getShows();
  }

  render() {
    const { shows } = this.props;

    return (
      <Grid container spacing={24}>
        {shows.map(show => (
          <GridCard item key={show.id} sm={12} md={6}>
            <SeriesCard show={show} />
          </GridCard>
        ))}
      </Grid>
    );
  }
}

export default SeriesPage;
