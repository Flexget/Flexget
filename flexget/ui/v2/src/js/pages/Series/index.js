import { connect } from 'react-redux';
import {
  GET_SHOWS,
} from 'store/series/shows/actions';
import { request } from 'utils/actions';
import Series from './Series';

export function mapStateToProps({ series }) {
  return {
    shows: series.shows.items,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    getShows: payload => dispatch(request(GET_SHOWS, payload)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Series);
export { Series };
