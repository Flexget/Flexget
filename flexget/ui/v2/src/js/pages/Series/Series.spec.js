import { mapStateToProps } from 'pages/Series';

describe('containers/series', () => {
  it('should return the right stuff', () => {
    expect(mapStateToProps({ series: {
      shows: {
        items: [],
      },
    } })).toMatchSnapshot();
  });
});
