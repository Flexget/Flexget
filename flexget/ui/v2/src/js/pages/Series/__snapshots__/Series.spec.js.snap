// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/series should return the right stuff 1`] = `
Object {
  "shows": Array [],
}
`;
